packages/dd-trace/src/scope/base.js

```diff
@@ -68,30 +68,33 @@ class Scope {
 
   _bindEmitter (emitter, span) {
     if (!this._isEmitter(emitter)) return emitter
-
-    wrapMethod(emitter, 'addListener', wrapAddListener, this, span)
-    wrapMethod(emitter, 'prependListener', wrapAddListener, this, span)
-    wrapMethod(emitter, 'on', wrapAddListener, this, span)
-    wrapMethod(emitter, 'once', wrapAddListener, this, span)
-    wrapMethod(emitter, 'prependOnceListener', wrapAddListener, this, span)
-
-    wrapMethod(emitter, 'removeListener', wrapRemoveListener)
-    wrapMethod(emitter, 'off', wrapRemoveListener)
-
-    wrapMethod(emitter, 'removeAllListeners', wrapRemoveAllListeners)
-
+    if (!emitter.__is_dd_emitter) {
+      Scope._wrapEmitter(emitter)
+    }
+    emitter.__dd_span = span
+    emitter.__dd_scope = this
     return emitter
   }
 
+  // Occaisionally we want to wrap a prototype rather than emitter instances,
+  // so we're exposing this as a static method. This lets gives us a faster
+  // path for binding instances of known EventEmitter subclasses.
+  static _wrapEmitter (emitter) {
```

Comment:
In order to not have this as private, for now the function could simply be added to the export (although the export will need to become an object).

Reply:
Eh, that's a much more impactful change, since this file is required in a whole bunch of places. I'd rather keep it as is. Having it as a static property here makes it even more clear what it's for, I think.